hpi-swt2/lecture-portal
# Feature/#147 add comprehension to lecture

# Resolves #147, #148, #149 

This PR adds the feature of lecture comprehension. Students can select one of three comprehension states. The lecturer sees an overview of the comprehension of the students for the lecture. After 10 minutes the state that the student selected is reset.

Acceptance Criteria:
- [x] 3 colored (red, yellow, green) buttons to switch the current level of comprehension (see mockup)
- [x] Submitting the current level of comprehension also submits a timestamp of when the information was submitted during the lecture
- [x] Updating the indicator adheres to soft real-time constraints
- [x] Indicator with 3-colored (red, yellow, green) sections, scaling proportionally to the number of students at the respective level of comprehension (see mockup) + number of voting students is displayed
- [x] If no students have voted so far, the indicator is grayed out
- [x] Updating the indicator adheres to soft real-time constraints
- [x] Votes that are older than 10 min are considered outdated and will not be counted (they behave like they were never recorded)
- [x] Student votes can be re-recorded by clicking the same color again, updating their timestamp to the current time
- [x] Annulation of outdated votes adheres soft-realtime constraints

Testing steps:
- log in as a student, join the lecture
- on the dashboard you should see a section called Comprehension Level
- select a level of comprehension by clicking on a circle
- after 10 minutes the circle is deselected
- you can change your comprehension level at any time
- log in as a lecturer 
- on the dashboard you should find a section called Comprehension Level 
- it should show an overview of the votes 

Steps needed for migration:
- bundle install

---

Definition Of Done:
